omx/gstomxtvp.c

```diff
@@ -0,0 +1,298 @@
+      GST_DEBUG_FUNCPTR (gst_omx_tvp_transform_ip);
+
+  GST_BASE_TRANSFORM_CLASS (klass)->set_caps =
+      GST_DEBUG_FUNCPTR (gst_omx_tvp_set_caps);
+
+  GST_BASE_TRANSFORM_CLASS (klass)->stop = GST_DEBUG_FUNCPTR (gst_omx_tvp_stop);
+
+  klass->cdata.component_name = "OMX.TI.VPSSM3.CTRL.TVP";
+  klass->cdata.core_name = "/usr/lib/libOMX_Core.so";
+  klass->cdata.hacks = GST_OMX_HACK_NO_COMPONENT_ROLE;
+
+  GST_DEBUG_CATEGORY_INIT (gst_omxtvp_debug, "omxtvp", 0,
+      "debug category for gst-omx tvp initializer");
+}
+
+/* tvp object init */
+static void
+gst_omxtvp_init (GstOMXTvp * self)
+{
+  self->standard = 1080;
+  self->scan_type = OMX_VIDEO_CaptureScanTypeProgressive;
+  self->mode_configured = FALSE;
+}
+
+static void
+gst_omx_tvp_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GstOMXTvp *self = GST_OMXTVP (object);
+  gchar *str_value;
+
+  switch (prop_id) {
+    case PROP_STANDARD:
+      self->standard = g_value_get_uint (value);
+      break;
+    case PROP_SCAN_TYPE:
+      str_value = g_value_dup_string (value);
+      if (!strcmp (str_value, "progressive")) {
+        self->scan_type = OMX_VIDEO_CaptureScanTypeProgressive;
+      } else if (!strcmp (str_value, "interlaced")) {
+        self->scan_type = OMX_VIDEO_CaptureScanTypeInterlaced;
+      } else {
+        GST_WARNING_OBJECT (self, "%s unsupported", str_value);
+        g_return_if_fail (0);
+      }
+      g_free (str_value);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+gst_omx_tvp_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec)
+{
+  GstOMXTvp *self = GST_OMXTVP (object);
+
+  switch (prop_id) {
+    case PROP_STANDARD:
+      g_value_set_uint (value, self->standard);
+      break;
+    case PROP_SCAN_TYPE:
+      if (self->scan_type == OMX_VIDEO_CaptureScanTypeProgressive)
+        g_value_set_string (value, "progressive");
+      else
+        g_value_set_string (value, "interlaced");
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/* GstBaseTransform vmethod implementations */
+
+static gboolean
+gst_omx_tvp_start (GstBaseTransform * base)
+{
+  GstOMXTvp *self = GST_OMXTVP (base);
+  GstOMXTvpClass *klass = GST_OMX_TVP_GET_CLASS (base);
+
+  self->comp =
+      gst_omx_component_new (GST_OBJECT_CAST (self), klass->cdata.core_name,
+      klass->cdata.component_name, klass->cdata.component_role,
+      klass->cdata.hacks);
+
+  if (!self->comp) {
+    GST_ERROR_OBJECT (self, "TVP component creation failed");
+    return FALSE;
+  }
+
+  if (!gst_omx_tvp_configure (self)) {
+    GST_ERROR_OBJECT (self, "TVP component configuration failed");
+    return FALSE;
+  }
+  return TRUE;
+}
+
+static GstFlowReturn
+gst_omx_tvp_transform_ip (GstBaseTransform * base, GstBuffer * outbuf)
```

Comment:
This is unnecessary. Just return FALSE in start if the TVP was not able to start. Don't implement this function.